src/table/components/head/HeadCellMenu.tsx

```diff
@@ -65,7 +69,7 @@ export default function HeadCellMenu({
         aria-controls={open ? 'sn-table-head-menu' : undefined}
         aria-expanded={open ? 'true' : undefined}
         aria-haspopup="true"
-        onClick={handleClickMenuOpen}
+        onClick={!open ? handleClickMenuOpen : handleClickMenuClose}
```

Comment:
This should be the same thing. 

```suggestion
        onClick={() => setMenuOpen(!menuOpen)}
```